static/js/popover_menus.js

```diff
@@ -185,4 +202,137 @@ export function initialize() {
             compose_enter_sends_popover_displayed = false;
         },
     });
+
+    delegate("body", {
+        ...default_popover_props,
+        target: ".actions_hover .zulip-icon-ellipsis-v-solid",
+        placement: "bottom",
+        popperOptions: {
+            modifiers: [
+                {
+                    // The placement is set to bottom, but if that placement does not fit,
+                    // the opposite top placement will be used.
+                    name: "flip",
+                    options: {
+                        fallbackPlacements: ["top"],
+                    },
+                },
+            ],
+        },
+        onShow(instance) {
+            on_show_prep(instance);
+            const $row = $(instance.reference).closest(".message_row");
+            const message_id = rows.id($row);
+            const args = popover_menus_data.get_actions_popover_content_args(message_id);
+            instance.setContent(parse_html(render_actions_popover_content(args)));
+            $row.addClass("has_popover has_actions_popover");
+            message_actions_popover_displayed = true;
+        },
+        onMount(instance) {
+            // We want click events to propagate to `instance` so that
+            // instance.hide gets called.
```

Comment:
There are no changes in how these events are processed other than remove `e.stopPropagation`.

Reply:
To be clear, we don't need them because TippyJS is doing those things already when setting up an `onMount` handler? Or something else?

Reply:
> the compose box also opens/closes when you click a popover item

Added a commit to fix it for the currently present tippy popovers.

For the click handlers inside popovers, we don't need to worry about them, since the popovers are appended directly to the body and don't propagate inside the app. Modified the comment too.

Reply:
Well, the class of issue would be the "main click handler" that's registered on the whole document running; but it already has explicit checks for popovers (I'm pretty sure it's the thing that closes the popover when clicking outside, and also it doesn't cancel compose for popover clicks), so the benefit for doing `stopPropagation` here would be a likely irrelevant performance optimization to not run this code:

```
    $(document).on("click", (e) => {                                                                              
        if (e.button !== 0 || $(e.target).is(".drag")) {                                                          
            // Firefox emits right click events on the document, but not on                                       
            // the child nodes, so the #compose stopPropagation doesn't get a                                     
            // chance to capture right clicks.                                                                    
            return;                                                                                               
        }                                                                                                         
                                                                                                                  
        // Dismiss popovers if the user has clicked outside them                                                  
        if (                                                                                                      
            $(                                                                                                    
                '.popover-inner, #user-profile-modal, .emoji-info-popover, .app-main [class^="column-"].expanded',
            ).has(e.target).length === 0                                                                          
        ) {                                                                                                       
            // Since tippy instance can handle outside clicks on their own,                                       
            // we don't need to trigger them from here.                                                           
            // This fixes the bug of `hideAll` being called                                                       
            // after a tippy popover has been triggered which hides                                               
            // the popover without being displayed.                                                               
            const not_hide_tippy_instances = true;                                                                
            popovers.hide_all(not_hide_tippy_instances);                                                          
        }                                                                                                         
                                                                                                                  
        if (compose_state.composing()) {                                                                          
            if (                                                                                                  
                $(e.target).closest("a").length > 0 ||                                                            
                $(e.target).closest(".copy_codeblock").length > 0                                                 
            ) {                                                                                                   
                // Refocus compose message text box if one clicks an external                                     
                // link/url to view something else while composing a message.                                     
                // See issue #4331 for more details.                                                              
                //                                                                                                
                // We do the same when copying a code block, since the                                            
                // most likely next action within Zulip is to paste it                                            
                // into compose and modify it.                                                                    
                $("#compose-textarea").trigger("focus");                                                          
                return;                                                                                           
            } else if (                                                                                           
                !window.getSelection().toString() &&                                                              
                // Clicking any input or text area should not close                                               
                // the compose box; this means using the sidebar                                                  
                // filters or search widgets won't unnecessarily close                                            
                // compose.                                                                                       
                !$(e.target).closest("input").length &&                                                           
                !$(e.target).closest("textarea").length &&                                                        
                !$(e.target).closest("select").length &&                                                          
                // Clicks inside an overlay, popover, custom                                                      
                // modal, or backdrop of one of the above                                                         
                // should not have any effect on the compose                                                      
                // state.                                                                                         
                !$(e.target).closest(".overlay").length &&                                                        
                !$(e.target).closest(".popover").length &&                                                        
                !$(e.target).closest(".micromodal").length &&                                                     
                !$(e.target).closest("[data-tippy-root]").length &&                                               
                !$(e.target).closest(".enter_sends").length &&                                                    
                $(e.target).closest("body").length                                                                
            ) {                                                                                                   
                // Unfocus our compose area if we click out of it. Don't let exits out                            
                // of overlays or selecting text (for copy+paste) trigger cancelling.                             
                // Check if the click is within the body to prevent extensions from                               
                // interfering with the compose box.                                                              
                compose_actions.cancel();                                                                         
            }                                                                                                     
        }                                                                                                         
    });                                                                                                           
```

Reply:
I think the conclusion is that we can go with this approach, but we should just understand why it's working correctly :)

Reply:
```js
 if (compose_state.composing()) {                                                                          
            if (                                                                                                  
                $(e.target).closest("a").length > 0 ||                                                            
                $(e.target).closest(".copy_codeblock").length > 0                                                 
            ) {    

```
above was the reason it is working correctly. Modified `on_show_prep` to accommodate for this:
```js
function on_show_prep(instance) {
    $(instance.popper).one("click", (e) => {
        // Don't do preventDefault since it leads to
        // direct links in popovers not working.
        e.stopPropagation();
        instance.hide();
        instance.destroy();
    });
    popovers.hide_all_except_sidebars(instance);
}
```

Reply:
I posted on the `on_show_prep` code with concerns about it, but I think the original approach of having the `stopPropagation` and hiding be explicit feels to me like a better design. Sometimes a bit of copy-paste is good.

That said, I think the main click handler should be skipping clicks inside these popovers due to this part: `!$(e.target).closest("[data-tippy-root]").length`, though I think it would be reasonable to add something that's explicitly for this popover module there.

Reply:
`$(e.target).closest("a").length > 0` is true when clicking on elements inside actions popovers atleast, so it never goes to the `else if` part.